packages/gel/src/options.ts

```diff
@@ -46,17 +46,15 @@ export interface SimpleRetryOptions {
 export type WarningHandler = (warnings: errors.GelError[]) => void;
 
 export const throwWarnings: WarningHandler = (warnings) => {
-  throw new Error(
-    `warnings occurred while running query: ${warnings.map((warn) => warn.message)}`,
-    { cause: warnings },
+  throw new AggregateError(
+    warnings,
+    `Warnings occurred while running query: ${warnings.map((warn) => warn.message)}`,
   );
 };
 
 export const logWarnings: WarningHandler = (warnings) => {
   for (const warning of warnings) {
-    console.warn(
-      new Error(`Gel warning: ${warning.message}`, { cause: warning }),
-    );
+    console.warn(new Error(`Gel warning: ${warning.message}`));
```

Comment:
How about just:

```suggestion
    console.warn(warning);
```

?

I guess maybe we want to indicate that this output is just some "warn"-level logging?

Maybe

```ts
export const logWarnings: WarningHandler = (warnings) => {
  if (warnings.length) {
    console.warn("Warning from Gel:");
  }
```

Reply:
I'm on the fence as well about whether we need to be explicit about this being a warning. We are calling `console.warn`, but that level isn't really represented in any default way. The message also starts with "QueryError: ..." (the type), so that is confusing.

I can't say I'm a fan of multiple subsequent console calls to facilitate line breaks. But I guess that could just be translated to
```ts
console.warn("Warning from Gel:\n" + warning.stack);
```
I'd want to confirm the `stack` has the correct `name`/`message` strings.

Reply:
Actually, I think this is my preferred solution: our default handlers collect all warnings into an `AggregateError` and either throw or `console.warn` that error with no further processing.

Reply:
Let me try and see what that looks like

Reply:
It doesn't look that good.
```ts
    const agg = new AggregateError(
      warnings,
      `warnings occurred while running query:\n${warnings
        .map((warn) => warn.message)
        .join('\n')}`,
    );
    console.warn(agg);
```
![Screenshot 2025-03-19 at 2 42 23 PM](https://github.com/user-attachments/assets/f6b9e11c-d566-4b60-9846-94c28a828eca)

Reply:
This seems fine to me, honestly. I personally don't much care for the duplication of the warnings' messages in the `AggegrateError.message`, but that ship has somewhat sailed.

Reply:
The these warnings all have the same effective stack trace for users. Why duplicate them & clutter the effective info?
```ts
const agg = new Error(
  `warnings occurred while running query:\n${warnings
    .map((warn) => warn.message)
    .join('\n')}`,
);
agg.name = '';
console.warn(agg);
```
![Screenshot 2025-03-19 at 2 50 38 PM](https://github.com/user-attachments/assets/bc844fe1-69fa-46e9-a9b6-9ec763aa8c3c)

Reply:
Honestly the duplicate stack trace with `cause` is how I get here in the first place.

Reply:
I mean, yeah, I agree with you: this is noisier than I would like, but it seems like the least opinionated way of doing it while also being consistent with the throwing version. 

I'm pretty on the fence here: the default `console`-based logging here is just pretty bad, and I think _that's_ really what feels frustrating here.

Alright, I guess I'm also fine with removing the stack traces like you're doing here and just concatenating all of the messages. Once we have time to figure out proper telemetry hopefully this will be easier to have a sane default.

Reply:
I agree that the console interface is poor. Actually I think what yall have with a configurable handler is best, and I'm already customizing for what fits me (that's different than what is here).
I was just trying to upstream my impression of a better DX default.

Reply:
yep, and I appreciate you taking the time to interate with me here! Let's take your approach of making a single new `Error` object with the warning messages concatenated into the message, and just `console.warn`ing that.

Reply:
Should I keep the line break and update tests or remove it?